Assets/Scripts/Models/Utility.cs

```diff
@@ -0,0 +1,784 @@
+    /// This is used to check what sprite to use if utility is next to each other.
+    /// </summary>
+    public bool LinksToNeighbour 
+    { 
+        get
+        {
+            return true;
+        }
+    }
+
+    /// <summary>
+    /// Gets the type of dragging that is used to build multiples of this utility. 
+    /// e.g walls.
+    /// </summary>
+    public string DragType 
+    { 
+        get
+        {
+            return "path";
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the parameters that is tied to the utility.
+    /// </summary>
+    public Parameter Parameters { get; private set; }
+
+    /// <summary>
+    /// Used to place utility in a certain position.
+    /// </summary>
+    /// <param name="proto">The prototype utility to place.</param>
+    /// <param name="tile">The base tile to place the utility on, The tile will be the bottom left corner of the utility (to check).</param>
+    /// <returns>Utility object.</returns>
+    public static Utility PlaceInstance(Utility proto, Tile tile)
+    {
+        if (proto.funcPositionValidation(tile) == false)
+        {
+            Debug.ULogErrorChannel("Utility", "PlaceInstance -- Position Validity Function returned FALSE. " + proto.Name + " " + tile.X + ", " + tile.Y + ", " + tile.Z);
+            return null;
+        }
+
+        // We know our placement destination is valid.
+        Utility obj = proto.Clone();
+        obj.Tile = tile;
+
+        if (tile.PlaceUtility(obj) == false)
+        {
+            // For some reason, we weren't able to place our object in this tile.
+            // (Probably it was already occupied.)
+
+            // Do NOT return our newly instantiated object.
+            // (It will be garbage collected.)
+            return null;
+        }
+
+        if (obj.LinksToNeighbour)
+        {
+            // This type of utility links itself to its neighbours,
+            // so we should inform our neighbours that they have a new
+            // buddy.  Just trigger their OnChangedCallback.
+            int x = tile.X;
+            int y = tile.Y;
+
+            for (int xpos = x - 1; xpos < x + 2; xpos++)
+            {
+                for (int ypos = y - 1; ypos < y + 2; ypos++)
+                {
+                    Tile tileAt = World.Current.GetTileAt(xpos, ypos, tile.Z);
+                    if (tileAt != null && tileAt.Utilities != null)
+                    {
+                        foreach (Utility utility in tileAt.Utilities.Values)
+                        {
+                            if (utility.Changed != null)
+                            {
+                                utility.Changed(utility);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // Call LUA install scripts
+
+        // TODO: Implement when EventActions.Trigger can take a Utility
+//        obj.EventActions.Trigger("OnInstall", obj);
```

Comment:
The issue is that EventActions do not work with utilities until it is generalized. I think is fine to leave the code in this case, since it will be implemented at some point and it will look exactly like that.